epoch.go

```diff
@@ -1861,13 +1870,20 @@ func (e *Epoch) proposeBlock(block VerifiedBlock) error {
 		return errors.New("failed to store block proposed by me")
 	}
 
+	// We might have received votes and finalizations from future rounds before we received this block.
```

Comment:
Why do we load the future messages before broadcasting the proposal? 

If we do it earlier then shouldn't we check if we have advanced to a higher round and if so, abort early and not broadcast the proposal?